lib/bolt/cli.rb

```diff
@@ -379,710 +341,584 @@ def validate(options)
+          result = app.secret_decrypt(options[:object], **options.slice(:plugin))
+          outputter.print_message(result)
+          SUCCESS
+        when 'encrypt'
+          result = app.secret_encrypt(options[:object], **options.slice(:plugin))
+          outputter.print_message(result)
+          SUCCESS
+        end
 
-    def show_targets
-      if options.keys.any? { |key| TARGETING_OPTIONS.include?(key) }
-        target_flag = true
-      else
-        options[:targets] = 'all'
+      when 'task'
+        case action
+        when 'run'
+          outputter.print_head
+          results = outputter.spin do
+            app.task_run(options[:object], options[:targets], **options.slice(:params))
+          end
+          rerun.update(results)
+          app.shutdown
+          outputter.print_summary(results, results.elapsed_time)
+          results.ok? ? SUCCESS : FAILURE
+        when 'show'
+          if options[:object]
+            outputter.print_task_info(**app.show_task(options[:object]))
+          else
+            outputter.print_tasks(**app.list_tasks(**options.slice(:filter)))
+          end
+          SUCCESS
+        end
       end
-
-      outputter.print_target_info(
-        group_targets_by_source,
-        inventory.source,
-        config.default_inventoryfile,
-        target_flag
-      )
     end
 
-    # Returns a hash of targets sorted by those that are found in the
-    # inventory and those that are provided on the command line.
+    # Process the target list by turning a PuppetDB query or rerun mode into a
+    # list of target names.
     #
-    private def group_targets_by_source
-      # Retrieve the known group and target names. This needs to be done before
-      # updating targets, as that will add adhoc targets to the inventory.
-      known_names = inventory.target_names
-
-      update_targets(options)
-
-      inventory_targets, adhoc_targets = options[:targets].partition do |target|
-        known_names.include?(target.name)
-      end
-
-      { inventory: inventory_targets, adhoc: adhoc_targets }
-    end
+    # @param pdb_client [Bolt::PuppetDB::Client] The PuppetDB client.
+    # @param rerun [Bolt::Rerun] The Rerun instance.
+    # @param options [Hash] The CLI options.
+    #
+    private def process_target_list(pdb_client, rerun, options)
+      return if options[:targets]
 
-    def list_groups
-      outputter.print_groups(inventory.group_names.sort, inventory.source, config.default_inventoryfile)
+      options[:targets] = if options[:query]
+                            pdb_client.query_certnames(options[:query])
+                          elsif options[:rerun]
+                            rerun.get_targets(options[:rerun])
+                          end
     end
 
-    # Looks up a value with Hiera as if in a plan outside an apply block, using
-    # provided variable values for interpolations
+    # TODO: Discuss whether we still want / need to collect analytics for
```

Comment:
I think so - it's useful when determining how urgent a bug is, or if we should continue to support bundled content (or bundle it).